chalk-parse/src/ast.rs

```diff
@@ -41,6 +42,34 @@ pub struct StructFlags {
     pub fundamental: bool,
 }
 
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct FnDefn {
+    pub name: Identifier,
+    pub parameter_kinds: Vec<ParameterKind>,
+    pub where_clauses: Vec<QuantifiedWhereClause>,
+    pub params: Vec<Field>,
+    pub returns: Vec<Field>,
```

Comment:
This doesn't feel right to me. If anything, it should be a single `Field`.

Reply:
Do you actually need the function params and return at all? I don't see where they should matter to Chalk :thinking:

Reply:
Well, we will *eventually* need this information. We are going to have to generate `FnMut` impls and the like.

Reply:
(But it might plausibly be done in a follow-up PR)